### PR TITLE
(iOS) Experimental pre-JSI synchronous SQLiteAdapter

### PR DESCRIPTION
**Pull request by radex**
- [SQLiteAdapter][iOS] Add new `synchronous` option to adapter: `new SQLiteAdapter({ ..., synchronous: true })`.
   When enabled, database operations will block JavaScript thread. Adapter actions will resolve in the
   next microtask, which simplifies building flicker-free interfaces

### CODE REVIEW

File: src/adapters/sqlite/index.js [hunk 1]
```diff
@@ -64,13 +81,79 @@ type NativeBridgeType = {
...
+  'setLocal',
+  'removeLocal',
+]
+
+type NativeBridgeType = {
+  // Async methods
+  ...NativeDispatcher,
+
+  // Synchronous methods
+  initializeSync?: (ConnectionTag, string, SchemaVersion) => SyncReturn<InitializeStatus>,
```
Comment (rozPierog) [inline src/adapters/sqlite/index.js]: I'm not a fan of `*Sync` naming scheme, it seems confusing especialy when we are talking about database that can `Synchronize` with server. Maybe something like `Synch` 🤷🏻‍♂️

File: src/adapters/sqlite/integrationTest.js [hunk 2]
```diff
@@ -1,18 +1,47 @@
...
-        const adapter = newAdapter()
+        const adapter = new SQLiteAdapter({ schema: testSchema, synchronous: isSynchronous })
+
+        if (isSynchronous) {
+          if (Platform.OS === 'ios') {
+            invariant(adapter._synchronous === true, 'this should be synchronous')
+          } else {
+            invariant(
+              adapter._synchronous === false,
+              'this should be asynchronous - android does not support synchronous adaoter',
```
Comment (rozPierog) [inline src/adapters/sqlite/integrationTest.js]:
```suggestion
              'this should be asynchronous - android does not support synchronous adapter',
```

Comment (rozPierog) [reply src/adapters/sqlite/integrationTest.js]: Also, we should avoid creating such discrepancy between Android and iOS. We all know how frustrating is when you are trying to implement something but for some reason it works only on one platform

Comment (radex) [reply src/adapters/sqlite/integrationTest.js]: True, OTOH: a) we agreed on Slack that it's not worth pursuing this internally on Android with current performance levels, b) it's an option you have to know about to use, c) there's a graceful fallback

File: src/adapters/sqlite/index.js (hunk 1 above)
Comment (radex) [reply src/adapters/sqlite/index.js]: riiiight. I'll go with `*Synchronous`

**Comment by radex** [comment]
fixed - let's wait for Travis to confirm i didn't mess anything up

**Comment by rozPierog** [comment]
Thanks 🎉

**Pull request merged**